target/linux/ath79/generic/config-default

```diff
@@ -8,7 +8,7 @@ CONFIG_LEDS_RESET=y
 CONFIG_MARVELL_PHY=y
 CONFIG_MICREL_PHY=y
 CONFIG_MTD_REDBOOT_PARTS=y
-CONFIG_MTD_REDBOOT_DIRECTORY_BLOCK=-2
+CONFIG_MTD_REDBOOT_DIRECTORY_BLOCK=-3
```

Comment:
Please separate commit as well, I would like to know, why do you need this change.

Reply:
This change is needed because that is the location of the directory block on this device. It's that simple. Ideally, this can even go in target/linux/generic.

Reply:
Ok, then please create separate commit for this ath79 kernel config changes and add there this description.

BTW what is going to happen on devices which rely on the current config `CONFIG_MTD_REDBOOT_DIRECTORY_BLOCK=-2` ? Are you making them unbootable?

Reply:
> Ok, then please create separate commit for this ath79 kernel config changes and add there this description.
> 
> BTW what is going to happen on devices which rely on the current config `CONFIG_MTD_REDBOOT_DIRECTORY_BLOCK=-2` ? Are you making them unbootable?

If -3 fails, -2 is tried, then -1. Under the current config, it starts with -2, then tries -1, and that's it. The current line in target/linux/generic would have it try -1 only. The ath25 target separately sets this to -3 like I'm doing. The only other place in OpenWrt where it is changed is that patch under ixp4xx where some board needs it to be -5.